oracle/weblogic-kubernetes-operator
# OWLS-84562 - added tests for Namespace management enhancements

Jenkins : https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2210/

Added tests to cover [https://jira.oraclecorp.com/jira/browse/OWLS-82046]  (https://github.com/oracle/weblogic-kubernetes-operator/pull/1860)
the tests don't cover Dedicated method, it is covered by other testclass
Testcases:
1. Create operator managing domain namespaces by SelectLabels
a. create namespace with specified label, make sure domain is started and operator can scale it
b. verify that domain with namespace with different label is not started/managed by operator, verify that as soon as namespace will be assigned the label, managed by operator, domain will be started and managed by operator.
c. verify that SelectLabel override domainNamespaces option from values and domain is not started even domainNamespace option is specified. 
d. verify that domain in default namespace is not started 
e. verify that operator can add/remove namespaces by upgrading helm chart and switch to different select method (RegExp)
f. verify that another operator (using List for domainNamespaces selection ) failed to share same namespace

2. Create operator managing domain namespaces by Expression searching for namespaces name starting with xoxoxo
a. create two namespace with namespaces staring with xoxoxo, make sure domains are started and operator can scale it
b. verify that domain with namespace with different name is not started/managed by operator
c. verify that RegExp override domainNamespaces option from values and domain is not started even domainNamespace option is specified. 
d. verify that domain in default namespace is not started 
e. verify that operator can add/remove namespaces by upgrading helm chart and switch to different select method (SelectLabels)
f. verify that another operator (using List for domainNamespaces selection ) failed to share same namespace

3. Create namespace before operator with setting enableRoleClusterBinding=false, and SelectLabel using that namespace, verify that domain can't be started , upgrade operator with reuse values, verify domain is started after upgrade.
